docs/griptape-framework/structures/task-memory.md

```diff
@@ -1,89 +1,320 @@
+```
+
+This is because the content of the webpage is too large to fit in the LLM's input token limit. We can fix this by storing the content in Task Memory, and then querying it with the `TaskMemoryClient`.
+
+```python
+from griptape.structures import Agent
+from griptape.tools import WebScraper, TaskMemoryClient
+
+agent = Agent(
+    tools=[
+        WebScraper(off_prompt=True),
+        TaskMemoryClient(off_prompt=False),
+    ]
+)
+
+agent.run(
+    "According to this page https://en.wikipedia.org/wiki/Elden_Ring, how many copies of Elden Ring have been sold?"
+)
+```
+
+And now we get the expected output:
+```
+[04/26/24 13:51:51] INFO     ToolkitTask 7aca20f202df47a2b9848ed7025f9c21
+                             Input: According to this page https://en.wikipedia.org/wiki/Elden_Ring, how many copies of Elden Ring have been sold?
+[04/26/24 13:51:58] INFO     Subtask 5b21d8ead32b4644abcd1e852bb5f512
+                             Thought: I need to scrape the content of the provided URL to find the information about how many copies of Elden Ring have been sold.
+                             Actions: [{"name": "WebScraper", "path": "get_content", "input": {"values": {"url": "https://en.wikipedia.org/wiki/Elden_Ring"}}, "tag":
+                             "scrape_elden_ring"}]
+[04/26/24 13:52:04] INFO     Subtask 5b21d8ead32b4644abcd1e852bb5f512
+                             Response: Output of "WebScraper.get_content" was stored in memory with memory_name "TaskMemory" and artifact_namespace
+                             "2d4ebc7211074bb7be26613eb25d8fc1"
+[04/26/24 13:52:11] INFO     Subtask f12eb3d3b4924e4085808236b460b43d
+                             Thought: Now that the webpage content is stored in memory, I need to query this memory to find the information about how many copies of Elden
+                             Ring have been sold.
+                             Actions: [{"tag": "query_sales", "name": "TaskMemoryClient", "path": "query", "input": {"values": {"memory_name": "TaskMemory",
+                             "artifact_namespace": "2d4ebc7211074bb7be26613eb25d8fc1", "query": "How many copies of Elden Ring have been sold?"}}}]
+[04/26/24 13:52:14] INFO     Subtask f12eb3d3b4924e4085808236b460b43d
+                             Response: Elden Ring sold 23 million copies by February 2024.
+[04/26/24 13:52:15] INFO     ToolkitTask 7aca20f202df47a2b9848ed7025f9c21
+                             Output: Elden Ring sold 23 million copies by February 2024.
+```
+
+## Sensitive Data
 
-prompt_driver = OpenAiChatPromptDriver(model="gpt-3.5-turbo")
+Because Task Memory splits up the storage and retrieval of data, you can use different models for each step.
+
+Here is an example where we use GPT-4 to orchestrate the Tools and store the data in Task Memory, and a self-hosted Llama-2 model to summarize the raw content.
+
+```python 
```

Comment:
This example may be good to copy into our Recipes section of our docs as well. I could see this being useful for people at a top-level example of the power of Griptape vs. needing to find it buried in our docs.